### PR TITLE
Feat: Added gfx::BufferFlag and gfx::DescriptorSetFlag

### PR DESCRIPTION
**Pull request by bluesky013**
Re: #

### Changelog

* Added gfx::BufferFlag and gfx::DescriptorSetFlag

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/gfx-base/GFXDescriptorSet.h [hunk 1]
```diff
@@ -78,8 +78,9 @@ class CC_DLL DescriptorSet : public GFXObject, public RefCounted {
     ccstd::vector<ObjectWithId<Buffer>> _buffers;
     ccstd::vector<ObjectWithId<Texture>> _textures;
     ccstd::vector<ObjectWithId<Sampler>> _samplers;
-
     bool _isDirty = false;
+    uint8_t _rsv[3] = {0};
```
Comment (star-e) [inline native/cocos/renderer/gfx-base/GFXDescriptorSet.h]: What does rsv mean? We might use the full name.

File: native/cocos/renderer/gfx-vulkan/VKBuffer.cpp [hunk 2]
```diff
@@ -73,18 +73,32 @@ void CCVKBuffer::createBufferView(uint32_t range) {
 }
 
 void CCVKBuffer::doDestroy() {
+    if (_gpuBuffer && !_isBufferView) {
+        CCVKDevice::getInstance()->gpuBarrierManager()->cancel(_gpuBuffer);
+        CCVKDevice::getInstance()->gpuBufferHub()->erase(_gpuBuffer);
+    }
+
+    if (_gpuBufferView && !hasFlag(_flags, BufferFlagBit::DISABLE_RESIZE)) {
```
Comment (star-e) [inline native/cocos/renderer/gfx-vulkan/VKBuffer.cpp]: I remember buffer view cannot be resized?

File: native/cocos/renderer/gfx-base/GFXDescriptorSet.h (hunk 1 above)
Comment (bluesky013) [reply native/cocos/renderer/gfx-base/GFXDescriptorSet.h]: I should use _padding.

File: native/cocos/renderer/gfx-vulkan/VKBuffer.cpp (hunk 2 above)
Comment (bluesky013) [reply native/cocos/renderer/gfx-vulkan/VKBuffer.cpp]: This is a whole range gpu buffer view of the `Buffer`.

File: native/cocos/renderer/gfx-base/GFXDescriptorSet.h (hunk 1 above)
Comment (star-e) [reply native/cocos/renderer/gfx-base/GFXDescriptorSet.h]: I don't think we need to do padding manually. We can assume developers understand c++ object alignment rules.
[c++ object](https://en.cppreference.com/w/cpp/language/object)
If we want to make sure object layout should meet certain requirements, we can use static_assert to check the layout.
For example, `static_assert(sizeof(DescriptorSet) == xxx);` etc.

**Pull request closed without merging**